File: Scala/src/main/scala/PrimaryClasses/K08_2sub.scala
package PrimaryClasses
import Golem.Golem

object AttackMethod extends Enumeration {
  type AttackMethod = Value

  val Attack: PrimaryClasses.AttackMethod.Value = Value(65)
  val Skill: PrimaryClasses.AttackMethod.Value = Value(99)
  val Magic: PrimaryClasses.AttackMethod.Value = Value(133)
}

object K08_2sub {
  def AttackGolem(lv: Int, playerHp: Int): Int ={
    var golem = new Golem(lv * 50 + 100, lv * 10 + 40, lv * 10 + 30)
    var _playerHp = playerHp

    printf("ゴーレムLv.%dが現れた！\n", lv + 1)
    while (golem.hp > 0) {
      printf("\n残りHP：%d\n", golem.hp)
      print("武器を選択してください（１．攻撃　２．特技　３．魔法）＞")
      val choice = scala.io.StdIn.readInt()
      var damage = 0



      choice match {
        case 1 => damage = AttackMethod.Attack.id
        case 2 => damage = AttackMethod.Skill.id
        case 3 => damage = AttackMethod.Magic.id
        case _ => ()
      }
      damage -= golem.defense

      if (damage <= 0) {
        damage = 0
        println("ゴーレム：「ハハハハハ、情けないな！貴様は弱すぎる！」")
        printf("ゴーレムがあなたを攻撃しました！攻撃値：%d\n", golem.attack)
        _playerHp = DamagePlayer(_playerHp, golem.attack)
        if (_playerHp <= 0) {
          _playerHp = 0
          println("あなたはゴーレムに負けました！ゲームオーバー！")
          return _playerHp
        }
        printf("あなたの残りHPは：%d\n", _playerHp)
      }
      printf("ダメージは%dです。\n", damage)
      golem.hp -= damage

      if (golem.hp <= 0)
        golem.hp = 0
      if (golem.hp == 0)
        printf("ゴーレムLv.%dを倒した！\n", lv + 1)
    }
    _playerHp
  }

  def DamagePlayer(playerHp: Int, attackPower: Int): Int = {
    playerHp - attackPower
  }
}

File: Java/src/com/ecc/gkgo1a/K08_2Sub.java
package com.ecc.gkgo1a;

import java.util.*;

class K08_2Sub {

    enum AttackMethod {
        Attack,
        Skill,
        Magic
    }

    static int AttackGolem(int lv, int playerHp) {
        Golem golem = new Golem(lv * 50 + 100, lv * 10 + 40, lv * 10 + 30);
        int _playerHp = playerHp;
        try
        {
            System.out.println(String.format("ゴーレムLv.%dが現れた！", lv + 1));
            while (golem.hp > 0)
            {
                System.out.println(String.format("\n残りHP：%d", golem.hp));
                System.out.print("武器を選択してください（１．攻撃　２．特技　３．魔法）＞");
                Scanner in = new Scanner(System.in);
                int choice = in.nextInt();
                int damage = 0;

                HashMap<AttackMethod, Integer> damageDispatcher = new HashMap<>();
                damageDispatcher.put(AttackMethod.Attack, 65);
                damageDispatcher.put(AttackMethod.Skill, 99);
                damageDispatcher.put(AttackMethod.Magic, 133);

                switch (choice)
                {
                    case 1:
                        damage = damageDispatcher.get(AttackMethod.Attack);
                        break;
                    case 2:
                        damage = damageDispatcher.get(AttackMethod.Skill);
                        break;
                    case 3:
                        damage = damageDispatcher.get(AttackMethod.Magic);
                        break;
                    default:
                            break;
                }

                damage -= golem.defense;
                if (damage <= 0)
                {
                    damage = 0;
                    System.out.println("ゴーレム：「ハハハハハ、情けないな！貴様は弱すぎる！」");
                    System.out.println(String.format("ゴーレムがあなたを攻撃しました！攻撃値：%d", golem.attack));
                    _playerHp = DamagePlayer(_playerHp, golem.attack);
                    if (_playerHp <= 0)
                    {
                        _playerHp = 0;
                        System.out.println("あなたはゴーレムに負けました！ゲームオーバー！");
                        return _playerHp;
                    }
                    System.out.println(String.format("あなたの残りHPは：%d", _playerHp));
                }
                System.out.println(String.format("ダメージは%dです。", damage));
                golem.hp -= damage;

                if (golem.hp <= 0)
                    golem.hp = 0;

                if (golem.hp == 0)
                    System.out.println(String.format("ゴーレムLv.%dを倒した！", lv + 1));
            }
        }
        catch (Exception ex)
        {
            System.err.println(ex.getMessage());
        }
        return _playerHp;
    }

    static int DamagePlayer(int playerHp, int attackPower) {
        return playerHp - attackPower;
    }
}
